tests/boards/espressif_esp32/cache_coex/README.rst

```diff
@@ -19,7 +19,7 @@ Make sure you have the ESP32 DevKitC connected over USB port.
 
 .. code-block:: console
 
-   west build -b esp32 tests/boards/espressif_esp32/cache_coex
+   west build -b esp32_devkitc_wrover tests/boards/espressif_esp32/cache_coex
```

Comment:
Such changes should be done together with the commit that split `esp32` into two separate platforms.

Reply:
As much I agree with you, it is quite hard to achieve good code reviewability and proper bisectionability at the same time with this type of braking changes bringing PRs. So I choose to group the commits by the folder they alter. 
However, I agree on reordering them so the sooner commits bring changes/files for the later one.

Reply:
> As much I agree with you, it is quite hard to achieve good code reviewability and proper bisectionability at the same time with this type of braking changes bringing PRs.

I don't agree here regarding bisectability/reviewability. IMO bisectability comes in pair with reviewability. If logical changes are kept minimal, then both the diff and the portion of code to be reviewed are minimal. In this PR however the logical changes (like renaming `CONFIG_SOC` to `CONFIG_SOC_SERIES`) are split across multiple commits. This fails to build (which means broken bisectability) and reviewing commit by commit is not applicable as well (all the commits need to be inspected in order to see if the change was complete).

> So I choose to group the commits by the folder they alter.

This is actually something that comes for free, without splitting commits. If you look at commit diff, then updated code chunks are sorted by the directory/file (file path). So grouping commits by folder gives no added value.

Reply:
I've checked the 'Contributor expectations' document under the 'Contribution guidelines' and you are right about that one! Will fix that.

Reply:
Breaking bisectability will affect a lot of developers, please do do not do that.